beacon/types/exec_header.go

```diff
@@ -48,33 +52,40 @@ func ExecutionHeaderFromJSON(forkName string, data []byte) (*ExecutionHeader, er
 	default:
 		return nil, fmt.Errorf("unsupported fork: %s", forkName)
 	}
-	if err := json.Unmarshal(data, obj); err != nil {
-		return nil, err
+
+	// Use a streaming decoder for efficiency, especially with large JSON payloads
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	if err := decoder.Decode(obj); err != nil {
```

Comment:
In the case that `data` is huge, and _at rest_, and being read from disk, I agree that it makes sense to use a streaming reader: it will read the data from disk a chunk at a time, thus using a low memory footprint. 

The case here, however, is that `data` is already a slice of data, in memory. What you are doing is adding a shim-layer which feeds bite-sized chunks to the json-reader. 

The only way in which I can imagine the new code to be more efficient or less memory-intensive, would be in case the `json` decoder internally copies the entire slice. In that case, the old code would utilize `2 x len(data)`, whereas the new code would only utilize `1 x len(data) + chunksize`.